openslides_backend/action/actions/user/import_.py

```diff
@@ -0,0 +1,118 @@
+from .update import UserUpdate
+from .user_mixin import DuplicateCheckMixin
+
+
+@register_action("user.import")
+class UserImport(DuplicateCheckMixin, ImportMixin):
+    """
+    Action to import a result from the action_worker.
+    """
+
+    model = ActionWorker()
+    schema = DefaultSchema(ActionWorker()).get_default_schema(
+        additional_required_fields={
+            "id": required_id_schema,
+            "import": {"type": "boolean"},
+        }
+    )
+    permission = OrganizationManagementLevel.CAN_MANAGE_USERS
+    skip_archived_meeting_check = True
+    import_name = "account"
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        instance = super().update_instance(instance)
+
+        # handle abort in on_success
+        if not instance["import"]:
+            return {}
+
+        # init duplicate mixin
+        data = self.result.get("rows", [])
+        for entry in data:
+            # Revert username-info and default-password-info
+            for field in ("username", "default_password"):
+                if field in entry["data"]:
+                    if field == "username" and "id" in entry["data"][field]:
+                        entry["data"]["id"] = entry["data"][field]["id"]
+                    entry["data"][field] = entry["data"][field]["value"]
+
+        search_data_list = [
+            {
+                field: entry["data"].get(field, "")
+                for field in ("username", "first_name", "last_name", "email")
+            }
+            for entry in data
+        ]
+        self.init_duplicate_set(search_data_list)
+
+        # Recheck and update data, update needs "id"
+        create_action_payload: List[Dict[str, Any]] = []
+        update_action_payload: List[Dict[str, Any]] = []
+        self.error = False
+        for payload_index, entry in enumerate(data):
+            if entry["state"] == ImportState.NEW:
+                if not entry["data"].get("username"):
+                    self.error = True
+                    entry["state"] = ImportState.ERROR
+                    entry["messages"].append("Error: could not find username")
+                elif self.check_username_for_duplicate(
+                    entry["data"]["username"], payload_index
+                ):
+                    self.error = True
+                    entry["state"] = ImportState.ERROR
+                    entry["messages"].append(
+                        "Error: want to create a new user, but username already exists."
+                    )
+                else:
+                    create_action_payload.append(entry["data"])
+            elif entry["state"] == ImportState.DONE:
+                search_data = self.get_search_data(payload_index)
+                if not entry["data"].get("username"):
+                    self.error = True
+                    entry["state"] = ImportState.ERROR
+                    entry["messages"].append("Error: could not find username")
```

Comment:
Better something like "Error: Want to update user, but missing username in import data"

Reply:
Done.